File: C_Sharp/p1431.cs

namespace p1431 {
    public class Solution {
        public IList<bool> KidsWithCandies(int[] candies, int extraCandies) {
            bool[] result = new bool[candies.Length];
            int max = int.MinValue;
            for(int i=0; i < candies.Length; i++){
                max = max < candies[i] ? candies[i] : max;
            }
            max -= extraCandies;
            for(int i=0; i < candies.Length; i++){
                result[i] = candies[i] >= max;
            }
            return result;
        }
    }
}

File: C_Plus_Plus/1001-2000/p1431.cpp
#include <vector>
using namespace std;

class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        int max = INT_MIN;
        vector<bool> result(candies.size(), false);
        for(int i=0; i<candies.size(); i++){
            max = max < candies[i] ? candies[i] : max;
        }
        max -= extraCandies;
        for(int i=0; i< result.size(); i++){
            result[i] = candies[i] >= max;
        }
        return result;
    }
};